### ISSUE TITLE
Proxy returns "HTTP/1.1 503 Service Unavailable"

### ISSUE THREAD

**Issue by BethanyJep** [ang3orge/trailhead-scraper#5]
Hello,
I have been trying to fetch trailhead data using the trailhead scraper, but when I want to extract the rank and awards data but it returns an error: Exception: Aura Action Error: Unable to tunnel through proxy. Proxy returns "HTTP/1.1 503 Service Unavailable"

**Comment by octatau** [comment]
@BethanyJep thanks for discovering this. I will work on fixing the issue shortly.

**Comment by cosm9747** [comment]
I also have the same issue as Bethany has for getting rank and award data (profile data works). From researching it, it looks to be an IP whitelisting issue. @ageonic, 1) any luck identifying / getting closer to fixing the issue and 2) did you have to go through any sort of IP whitelisting process to access those endpoints yourself?

I could be off base here, but the API call for rank and awards don't seem to be fully public like the profile one is

**Comment by conorsmithdeloitte** [comment]
@ageonic, if you could confirm that you don't get this service unavailable error when testing the rank or award endpoints from your end, that would be very helpful for me to figure out how I can diagnose the error for myself. Thank you :)

**Comment by octatau** [comment]
@BethanyJep @cosm9747 @conorsmithdeloitte 
It appears that Trailhead has shifted to using GraphQL to retrieve award/rank data. Would any of you be interested in helping restructure this project to incorporate the new endpoints?

**Comment by conorsmithdeloitte** [comment]
@ageonic, yes, I'd be interested in helping restructure the project to incorporate the new endpoints. I'd just need some more direction from you on next steps. Would be helpful to get on a call sometime in the coming weeks to further discuss, or if you have any documentation you could send over that would help w/ updating the API that works as well. Thanks!

**Comment by conorsmithdeloitte** [comment]
@ageonic, I have another colleague that can help as well. Let me know next steps!

**Comment by conorsmithdeloitte** [comment]
@ageonic, quick follow-up here. Let me know if the solution you suggested above about restructuring for GraphQL will work to get badge data and if you could give some more details of how the endpoints need to be updated to account for the shift to GraphQL. Thanks

**Comment by octatau** [comment]
@conorsmithdeloitte I apologize for the delay in responding. Do you mind moving this conversation to Slack? Let me know your (and your colleague's) email address and I can invite you to a workspace.
@BethanyJep @cosm9747 feel free to send me your email addresses as well if you would like to be part of the conversation.

**Comment by AJCcs** [comment]
Hi @ageonic , I would like to contribute, how can we contact us?
